ci: setup ci and fix tests

dido18 opened the pull request:
Add the `go.mod` file 
Remove the `io/ioutil` deprecated package
Add the `githubacionts` to check and test

NOTE: the tests are executed inside a docker container since the `apt upgrade`  requires root access. After merging this pr the tests can be refactored to use goldenfiles

dido18 committed:
Refactor file reading to use os package instead of ioutil

dido18 committed:
Add Taskfile and .gitignore; improve error messages and test cleanup

dido18 committed:
Add GitHub Actions workflow for running Go tests

dido18 committed:
Update GitHub Actions workflow to trigger on all branches

dido18 committed:
Remove unnecessary dependency installation step and revert Go version to 1.24.0

dido18 committed:
Update TestSearch to use "bash" for consistent test results

dido18 committed:
Fix test

dido18 committed:
fix test using go-cmp

dido18 committed:
refactor test cleanup to simplify deferred removal of test files

dido18 committed:
WIP: disabled failing test

cmaglie commented:
We have a template for the go-checks: https://github.com/arduino/tooling-project-assets/blob/main/workflow-templates/check-go-task.md

and go-tests: https://github.com/arduino/tooling-project-assets/blob/main/workflow-templates/test-go-task.md

dido18 committed:
added gitkeep

dido18 committed:
Cross linux test (#1)


---------

Co-authored-by: giulio93 <pilotto.giulio@gmail.com>

dido18 committed:
Add GitHub Actions workflow for Go checks and update Taskfile with linting and formatting tasks

dido18 committed:
Update GitHub Actions workflow and add GolangCI configuration

- Refactor Go test workflow to use environment variable for Go version
- Add .golangci.yaml for linting and formatting configurations
- Clean up commented debug statements in repos.go
- Rename parameter in EditRepository function for clarity
- Adjust file permissions in replaceFile function

dido18 committed:
Refactor Taskfile to streamline Go testing and remove deprecated linter setup

dido18 committed:
Add GitHub Actions workflow for Go testing and coverage reporting

dido18 committed:
Update CI configuration to specify repository for coverage reporting and enhance README with status badges

dido18 committed:
actions badge

dido18 committed:
Add GitHub Actions workflow for Go testing and coverage reporting
